tools/tsp-client/src/commands.ts

```diff
@@ -567,3 +570,23 @@ export async function sortSwaggerCommand(argv: any): Promise<void> {
   await writeFile(swaggerFile, JSON.stringify(sorted, null, 2));
   Logger.info(`${swaggerFile} has been sorted.`);
 }
+
+
+function getEmitterPackageJsonPath(repoRoot: string, tspLocation: TspLocation): string {
+  const relativePath = tspLocation.emitterPackageJsonPath ?? defaultRelativeEmitterPackageJsonPath;
+  return joinPaths(repoRoot, relativePath);
+}
+
+function getEmitterLockPath(emitterPackageJsonPath: string): string {
+  const emitterPackageJsonFileName = path.basename(emitterPackageJsonPath, path.extname(emitterPackageJsonPath));
```

Comment:
Why are we including the extension here?

Reply:
This computes the file name from the file path + ext.

Reply:
I thought basename with the emitterPackageJsonPath would work without you needing to specify the extension? Is that not so?

Reply:
I believe it includes the extension.